packages/hardhat-core/src/types/config.ts

```diff
@@ -120,6 +121,7 @@ export interface HardhatNetworkConfig {
   initialDate: string;
   loggingEnabled: boolean;
   forking?: HardhatNetworkForkingConfig;
+  coinbase?: string;
```

Comment:
Why optional here? I guess I'm still trying to wrap my head around why some fields here should be optional any some shouldn't.  I thought after the #1666 work we wanted to minimize the optionality on this type.

Reply:
This is something I struggled a bit with too. I think that to keep backward compatibility new fields to existing types need to be optional. I kind of hate it, but otherwise existing code may stop working.